<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>588</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1000</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>PCIE Demo</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_8">
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_3">
      <item>
       <widget class="QGroupBox" name="groupBoxControl">
        <property name="title">
         <string>设备控制</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayoutDevice">
           <item>
            <widget class="QLabel" name="labelDeviceSelect">
             <property name="maximumSize">
              <size>
               <width>60</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>设备选择：</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboDevices"/>
           </item>
           <item>
            <widget class="QPushButton" name="btnRefreshDevices">
             <property name="maximumSize">
              <size>
               <width>60</width>
               <height>16777215</height>
              </size>
             </property>
             <property name="text">
              <string>刷新设备</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QGridLayout" name="gridLayout">
           <item row="4" column="4">
            <widget class="QPushButton" name="btnStart">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>开始</string>
             </property>
            </widget>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_2">
             <property name="text">
              <string>同步控制：</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_2">
             <item>
              <widget class="QPushButton" name="btnHardwareSyncEnable">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>进入同步模式</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnHardwareSyncOpen">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>开启</string>
               </property>
               <property name="checkable">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>发送模式：</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QLineEdit" name="editSendDMASize">
                             <property name="text">
                                <string>1024</string>
                             </property>
                             <property name="toolTip">
                                <string>每次 DMA 传输大小（KB）。推荐 1024 KB（1 MB）或更大以提高吞吐。</string>
                             </property>
                             <property name="placeholderText">
                                <string>推荐：1024 (KB)</string>
                             </property>
            </widget>
           </item>
               <item row="1" column="1">
                <widget class="QCheckBox" name="chkUseMemorySource">
                  <property name="text">
                   <string>内存生成（测试）</string>
                  </property>
                  <property name="toolTip">
                   <string>使用内存生成作为输入源以测试 PCIe 管道吞吐（无需文件）</string>
                  </property>
                </widget>
               </item>
                      <item row="1" column="2">
                        <widget class="QCheckBox" name="chkUseShmSource">
                           <property name="text">
                            <string>共享内存（SHM）</string>
                           </property>
                           <property name="toolTip">
                            <string>从 POSIX 共享内存环读取数据作为输入源</string>
                           </property>
                        </widget>
                      </item>
                <item row="1" column="3">
                 <widget class="QLabel" name="labelInputSource">
                  <property name="text">
                   <string>输入源：文件</string>
                  </property>
                  <property name="toolTip">
                   <string>显示当前选择的输入源：文件或内存生成（测试）</string>
                  </property>
                 </widget>
                </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>板卡控制：</string>
             </property>
            </widget>
           </item>
           <item row="0" column="3">
            <widget class="QLabel" name="labelSendBufferSize">
             <property name="text">
              <string>缓冲区大小(MB)：</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout">
             <item>
              <widget class="QPushButton" name="btnOpen">
               <property name="text">
                <string>打开</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnClose">
               <property name="enabled">
                <bool>false</bool>
               </property>
               <property name="text">
                <string>关闭</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="0">
            <widget class="QLabel" name="labelSendDMASize">
             <property name="text">
              <string>DMA大小(KB)：</string>
             </property>
            </widget>
           </item>
           <item row="2" column="4">
            <widget class="QComboBox" name="comboSpeed">
             <property name="currentText">
              <string>4Hz</string>
             </property>
             <item>
              <property name="text">
               <string>4Hz</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>2Hz</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>1Hz</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>0.5Hz</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>0.25Hz</string>
              </property>
             </item>
            </widget>
           </item>
           <item row="2" column="3">
            <widget class="QLabel" name="labelSpeed">
             <property name="text">
              <string>发送速率：</string>
             </property>
            </widget>
           </item>
           <item row="0" column="2">
            <spacer name="horizontalSpacer_5">
             <property name="orientation">
              <enum>Qt::Orientation::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item row="2" column="1">
            <layout class="QHBoxLayout" name="horizontalLayout_5">
             <item>
              <widget class="QRadioButton" name="radioSendMode1">
               <property name="text">
                <string>列表单次</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="radioSendMode2">
               <property name="text">
                <string>列表循环</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="0" column="4">
            <widget class="QLineEdit" name="editSendBufferSize">
                             <property name="text">
                                <string>64</string>
                             </property>
                             <property name="toolTip">
                                <string>发送缓存总大小（MB）。推荐 64 MB 起点以聚合多个 DMA 请求。</string>
                             </property>
                             <property name="placeholderText">
                                <string>推荐：64 (MB)</string>
                             </property>
            </widget>
           </item>
           <item row="4" column="3">
            <widget class="QPushButton" name="btnReset">
             <property name="enabled">
              <bool>false</bool>
             </property>
             <property name="text">
              <string>复位</string>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="SpeedChartWidget" name="widgetOscope">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QGroupBox" name="groupBoxFile">
        <property name="title">
         <string>待发送文件列表</string>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_5">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_7">
             <item>
              <widget class="QPushButton" name="btnFolderSelect">
               <property name="maximumSize">
                <size>
                 <width>50</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>选择</string>
               </property>
              </widget>
             </item>
             <item>
              <spacer name="horizontalSpacer">
               <property name="orientation">
                <enum>Qt::Orientation::Horizontal</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>0</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item>
              <widget class="QPushButton" name="btnUpSelectFile">
               <property name="enabled">
                <bool>true</bool>
               </property>
               <property name="maximumSize">
                <size>
                 <width>36</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>↑</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnDownSelectFile">
               <property name="maximumSize">
                <size>
                 <width>36</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>↓</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnDeleteAllFiles">
               <property name="maximumSize">
                <size>
                 <width>40</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="btnDeleteSelectedFile">
               <property name="maximumSize">
                <size>
                 <width>40</width>
                 <height>16777215</height>
                </size>
               </property>
               <property name="text">
                <string>删除</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <widget class="MyListWidget" name="listFiles">
             <property name="minimumSize">
              <size>
               <width>400</width>
               <height>0</height>
              </size>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyListWidget</class>
   <extends>QListWidget</extends>
   <header>mylistwidget.h</header>
  </customwidget>
  <customwidget>
   <class>SpeedChartWidget</class>
   <extends>QWidget</extends>
   <header>speedchartwidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
